front/app/containers/Admin/projects/project/files/components/FilePreview/CsvFilePreview/index.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useEffect, useState, useRef } from 'react';
+
+import {
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  Spinner,
+  Text,
+  Box,
+} from '@citizenlab/cl2-component-library';
+import Papa from 'papaparse';
+import styled from 'styled-components';
+
+import { useIntl } from 'utils/cl-intl';
+
+import DownloadFileButton from '../../DownloadFileButton';
+import messages from '../messages';
+
+const MAX_ROWS = 50; // Default max rows to read from the CSV (helps with performance)
+const MAX_FILE_SIZE_MB = 2; // Max file size in MB we will allow previews for (helps with performance)
+
+type Props = {
+  url: string;
+  fileSize?: number; // in bytes
+};
+
+const ScrollableTableContainer = styled(Box)`
+  transform: rotateX(
+    180deg
+  ); /* Rotation to allow the X scroll bar on the top */
+  cursor: grab;
```

Comment:
Slightly unconventional approach here, but I really wanted the scroll bar on the top here for better UX + visibility that this was horizontally scroll-able. Found this solution online by applying `rotateX` and thought it was clever. Let me know if you have any concerns though :)